### PR TITLE
set a new func for TQDepth

### PR DESCRIPTION
**Pull request by avelsh**
add a new feuture to count the TQDepth for TwoQubitEncoding

### CODE REVIEW

File: src/operations/Operation.cpp [hunk 1]
```diff
@@ -249,4 +249,29 @@ void Operation::addDepthContribution(std::vector<std::size_t>& depths) const {
...
+
+  std::size_t maxDepth = 0;
+  for (const auto& target : getTargets()) {
+    maxDepth = std::max(maxDepth, depths[target]);
+  }
+  for (const auto& control : getControls()) {
+    maxDepth = std::max(maxDepth, depths[control.qubit]);
+  }
+
+  if (!getControls().empty()) {
```
Comment (pehamTom) [inline src/operations/Operation.cpp:266]: There are two-qubit gates that do not have a control (e.g. SWAP gate) but are still two-qubit gates.

File: src/operations/Operation.cpp [hunk 2]
```diff
@@ -249,4 +249,29 @@ void Operation::addDepthContribution(std::vector<std::size_t>& depths) const {
   }
 }
 
+void Operation::addTwoQubitDepthContribution(
```
Comment (pehamTom) [inline src/operations/Operation.cpp:252]: This doesn't just count two-qubit gates but any gate with a control. So a multi-controlled x-gate would still be considered a two-qubit gate in this case. I actually don't think this is a huge problem but the feature and the methods should be renamed accordingly (i.e. multiQubitGateDepth or something).

File: include/operations/Operation.hpp [hunk 3]
```diff
@@ -135,6 +135,8 @@ class Operation {
   }
 
   virtual void addDepthContribution(std::vector<std::size_t>& depths) const;
+  virtual void
+  addTwoQubitDepthContribution(std::vector<std::size_t>& depths) const;
```
Comment (pehamTom) [inline include/operations/Operation.hpp:139]: This feature doesn't work correctly for a `CompoundOperation`. For these the method would have to be overridden so as to recursively compute the depth.

**Review by pehamTom — CHANGES_REQUESTED** [review]
Thank you for this feature. In addition to the changes below, please add some test cases to check if everything works correctly.

**Comment by burgholzer** [comment]
Closing this for now, as no one seems to be actively working on it.
Feel free to re-open if this changes at any point in time.

**Pull request closed without merging**